### PR TITLE
TCGC, add crossLanguageDefinitionId to SdkClientType

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Fix https://github.com/Azure/typespec-azure/issues/954

### CODE REVIEW

File: packages/typespec-client-generator-core/src/package.ts [hunk 1]
```diff
@@ -522,6 +522,10 @@ function createSdkClientType<TServiceOperation extends SdkServiceOperation>(
     // eslint-disable-next-line deprecation/deprecation
     arm: client.kind === "SdkClient" ? client.arm : false,
     decorators: diagnostics.pipe(getTypeDecorators(context, client.type)),
+    // if it is client, the crossLanguageDefinitionId is the namespace.TestClient, if it is operation group, the crosslanguageDefinitionId is the namespace.Test
+    crossLanguageDefinitionId: isClient
```
Comment (haolingdong-msft) [inline packages/typespec-client-generator-core/src/package.ts:526]: if it's `SdkClient`, the logic will follow `listClients` logic: https://github.com/Azure/typespec-azure/blob/2ce5dac0f866860f31457fad551e457ba34ad11c/packages/typespec-client-generator-core/src/decorators.ts#L338

If it's operation group, the `crossLanguageDefinitionId` will be the same as `crossLanguageDefinitionId` for `clientaccessor`, [logic](https://github.com/Azure/typespec-azure/blob/2ce5dac0f866860f31457fad551e457ba34ad11c/packages/typespec-client-generator-core/src/package.ts#L407)

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/typespec-client-generator-core/src/package.ts (hunk 1 above)
Comment (haolingdong-msft) [reply packages/typespec-client-generator-core/src/package.ts:526]: I might leave this question to Chenjie/Izzy, I checked the existing behavior, crossLanguageDefinitionId for client is `namespace.clientName`, so I just keep it consistent.

Comment (tadelesh) [reply packages/typespec-client-generator-core/src/package.ts:526]: i think we should keep the original ns name and change the client's cxlangdefid.

Comment (haolingdong-msft) [reply packages/typespec-client-generator-core/src/package.ts:526]: @tadelesh I did not quite get your point. Could you elaborate more using the test cases?

File: packages/typespec-client-generator-core/src/package.ts [hunk 2]
```diff
@@ -522,6 +522,8 @@ function createSdkClientType<TServiceOperation extends SdkServiceOperation>(
     // eslint-disable-next-line deprecation/deprecation
     arm: client.kind === "SdkClient" ? client.arm : false,
     decorators: diagnostics.pipe(getTypeDecorators(context, client.type)),
+    // if it is client, the crossLanguageDefinitionId is the namespace.TestClient, if it is operation group, the crosslanguageDefinitionId is the namespace.Test
```
Comment (tadelesh) [inline packages/typespec-client-generator-core/src/package.ts:525]: could you change this comment?

**Review by tadelesh — APPROVED** [review]
@haolingdong-msft please wait to merge until @iscai-msft to confirm the change of client cross language definition id.

**Review by iscai-msft — APPROVED** [review]

**Comment by haolingdong-msft** [comment]
> @haolingdong-msft please wait to merge until @iscai-msft to confirm the change of client cross language definition id.

sure, @tadelesh  I will leave it to you or @iscai-msft  to  merge the pr. Thanks

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**